resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala

```diff
@@ -243,7 +243,7 @@ private[yarn] class ExecutorRunnable(
 
     // Add log urls
     container.foreach { c =>
-      sys.env.filterKeys(_.endsWith("USER")).foreach { user =>
+      sys.env.get("SPARK_USER").orElse(sys.env.get("USER")).foreach { user =>
```

Comment:
So our preference would be `SPARK_USER` -> `USER`, right? I have to apply this to #23260 given that I'm refactoring the this code block in #23260.

Reply:
Maybe explicitly set the `SPARK_USER` env variable to the value of `Utils.getCurrentUserName`? Instead of trying to figure out what's the correct value from the env. (`USER` could be wrong if you have kerberos, for example.)